metomi/rose/job_runner.py

```diff
@@ -17,9 +17,17 @@
 """A multiprocessing runner of jobs with dependencies."""
 
 import asyncio
+
 from metomi.rose.reporter import Event
 
 
+# set containing references to "background" coroutines that are not referenced
+# from any code (i.e. are not directly awaited), adding them to this list
+# avoids the potential for garbage collection to delete them whilst they are
+# running
+_BACKGROUND_TASKS = set()
```

Comment:
This ensures that background tasks are always referenced whilst they are running.

That is so long as the module doesn't get housekept :see_no_evil: 

For details: https://docs.python.org/3/library/asyncio-task.html#asyncio.create_task